### PR TITLE
Purge command

### PR DESCRIPTION
**Pull request by caouecs**
Hello,
we created the purge command to delete all packages not needed by our new versions, and our tests. It based on the last archive file.

### CODE REVIEW

**Comment by raistlin** [comment]
Any chance this will be merged?

File: README.md [hunk 1]
```diff
@@ -13,9 +13,17 @@ Usage
...
 
-Read the more detailed instructions in the 
+Read the more detailed instructions in the
 [documentation](http://getcomposer.org/doc/articles/handling-private-packages-with-satis.md).
 
+Purge
+-----
+
+If you choose to archive packages in your server, you can have useless files.  
+With the `purge` command, you delete these files.
```
Comment (till) [inline README.md]: :+1: 

I think this is a tricky feature to add.

**Comment by GromNaN** [comment]
:+1:

**Comment by Divi** [comment]
:+1:

**Comment by alcohol** [comment]
How about adding an option/flag to specify how many builds to keep, e.g. to keep the last 3.

**Comment by ro0NL** [comment]
@alcohol any updates on this? I think it's definitely a nice-to-have feature and should be reviewed/merged or at least decline the pull request otherwise...

**Comment by alcohol** [comment]
There you go.

**Comment by ro0NL** [comment]
That was fast :astonished: Thanks.

**Comment by ro0NL** [comment]
@alcohol im not sure we're going to run this in production. Im testing it locally and everything got removed from `dist/` ;-)

Im debugging it right now.. and as far as I can see the regex at https://github.com/composer/satis/blob/master/src/Composer/Satis/Command/PurgeCommand.php#L100,L110 is totally failing :exclamation: 

Our results:
```php
$needed = array('vendor-some-bundle');
$files = array('vendor-some-bundle-v1.0.0-XXXXXX.zip')
```

So basicly the regex is applied as follow https://regex101.com/r/lF5mI7/1

I.e. `$matches[1]` is set to `vendor-some` whish is not in the `$needed` array, so unlink is called. However it actually "needed".

So the regex vs. package naming is problematic due dashes I guess. Also it has no start/end modifiers (`^...$`) which can be a potential risk/another false positive.

Im testing our usecase, and planning to fix it at regex-level.

For your information ;-)

**Comment by ro0NL** [comment]
@alcohol following regex seems OK: `/^(.+)-(?:[^-]+)-(?:[^-]+)\.(?:.+)$/`

https://regex101.com/r/eW2aY8/1

Can u fix this in master or should i go with a PR?

**Comment by ro0NL** [comment]
Also `include/` is never cleaned up.. which is a minor for now. I guess these can all be removed except the one referenced in `package.json`, right?

**Comment by yoshz** [comment]
Please keep in mind that when you purge the dist files, any references in composer.lock of your project will still point to those files!

**Comment by alcohol** [comment]
@ro0NL I might have a looksie tomorrow. But a PR is always welcome, so if you have the time, please feel free to submit one.

**Pull request merged**